### PR TITLE
Feature/UI improvement issue621

### PR DESCRIPTION
**Pull request by pedrule**

### CODE REVIEW

File: ui/agnostic_components/virtual-scroll/src/VirtualScrollBase.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
+import { LitElement, property, css, CSSResultArray } from "lit-element";
+import { ScrollMixin } from "./behaviors/ScrollMixin";
+import { applyMixins } from "./behaviors/Mixin.model";
+import { InstanceMixin } from "./behaviors/InstanceMixin";
+import { DomMixin } from "./behaviors/DomMixin";
+import { ContainerMixin } from "./behaviors/ContainerMixin";
```
Comment (ideoma) [inline ui/agnostic_components/virtual-scroll/src/VirtualScrollBase.ts:6]: This file is missing

File: ui/src/scenes/Result/index.tsx [hunk 2]
```diff
@@ -181,6 +189,7 @@ const Result = () => {
       </Menu>
 
       <Content>
+        <virtual-scroll>tessst</virtual-scroll>
```
Comment (ideoma) [inline ui/src/scenes/Result/index.tsx:192]: This looks incomplete.
The purpose of this feature is to virtualize scrolling of items inside `scenes/Schema`

File: ui/README.md [hunk 3]
```diff
@@ -21,3 +21,15 @@ Run the command:
 ```
 npm run build
 ```
+
```
Comment (ideoma) [inline ui/README.md:24]: Would be good if we can build and run node code completely without docker.
Can you update instructions on how to build the UI code? At the moment it is built at Azure pipeline on ever PR by running `npm run build` and I think but this PR will break the build,

**Review by ideoma — CHANGES_REQUESTED** [review]

File: ui/README.md (hunk 3 above)
Comment (marregui) [reply ui/README.md:24]: hi @pedrule,

I went straight to the README.md for instructions and followed what it says:
1. `npm install`
```
npm install
npm ERR! code ENOTFOUND
npm ERR! syscall getaddrinfo
npm ERR! errno ENOTFOUND
npm ERR! network request to http://host.docker.internal:4873/@questdb_design-system%2fvirtual-scroll failed, reason: getaddrinfo ENOTFOUND host.docker.internal
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/marregui/.npm/_logs/2021-09-07T09_56_17_284Z-debug.log
```
that did not work, so I tried to run:

2. `docker-compose up`:
```
 => ERROR [7/7] RUN npm i                                                                                                                                                                                                                3.5s
------
 > [7/7] RUN npm i:
#12 0.327 npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
#12 3.480 npm ERR! code E404
#12 3.481 npm ERR! 404 Not Found - GET https://registry.npmjs.org/@questdb_design-system%2fvirtual-scroll - Not found
#12 3.481 npm ERR! 404
#12 3.481 npm ERR! 404  '@questdb_design-system/virtual-scroll@0.0.1' is not in the npm registry.
#12 3.481 npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
#12 3.481 npm ERR! 404 It was specified as a dependency of 'src'
#12 3.481 npm ERR! 404
#12 3.481 npm ERR! 404 Note that you can also install from a
#12 3.481 npm ERR! 404 tarball, folder, http url, or git url.
#12 3.483
#12 3.483 npm ERR! A complete log of this run can be found in:
#12 3.483 npm ERR!     /root/.npm/_logs/2021-09-07T09_46_49_476Z-debug.log
------
executor failed running [/bin/sh -c npm i]: exit code: 1
ERROR: Service 'ui' failed to build : Build failed
```

Could you possibly amend the instructions in the README.md file?. Ideally for both local install/run from the branch, and using docker.

Thank you in advance!!

PS. I did try to run this, but obviously the install failed, so there are missing libs:
```
npm run build
> questdb-console@0.0.1 build
> cross-env NODE_ENV=production webpack && node utils/deploy.js
chunk    {0} qdb.css, qdb.js (main) 4.33 MiB [entry] [rendered]
ERROR in ./src/scenes/Result/index.tsx
Module not found: Error: Can't resolve '@questdb_design-system/virtual-scroll' in '/Users/marregui/workspace/TEMP/questdb/ui/src/scenes/Result'
resolve '@questdb_design-system/virtual-scroll' in '/Users/marregui/workspace/TEMP/questdb/ui/src/scenes/Result'
  Parsed request is a module
  using description file: /Users/marregui/workspace/TEMP/questdb/ui/package.json (relative path: ./src/scenes/Result)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
...
```

**Pull request closed without merging**